### PR TITLE
Add Task to Trigger Jenkins Build using Tekton

### PR DESCRIPTION
**Pull request by vinamra28**
# Changes

The following task can be used to trigger an existing Jenkins pipeline from Tekton using the CURL request by providing the required parameters.

Signed-off-by: vinamra28 <vinjain@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Yaml file complies with [yamllint](https://github.com/adrienverge/yamllint) rules.
- [X] Complies with [Catalog Orgainization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: trigger-jenkins-pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+## Parameters
+
+- **JENKINS_HOST_URL**: The URL on which Jenkins is running (**Required**)
+- **JOB_NAME**: The Job name which needs to be triggered (**Required**)
+- **JENKINS_SECRETS**: The name of the secret containing the username and API token for authenticating the Jenkins (_Default_: jenkins-credentials) (**Required**)
+- **ARGS**: Extra parameters which needs to be appended in the `CURL` request. (_Default_: ""). `ARGS` is of type `array` so multiple arguments can be appended. `ARGS` can be provided as follows:-
+
+  ```yaml
+  params:
+    - name: ARGS
```
Comment (siamaksade) [inline trigger-jenkins-pipeline/README.md:22]: Might be more suitable to call this param `JOB_PARAMS`

File: trigger-jenkins-pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      value: |
```
Comment (siamaksade) [inline trigger-jenkins-pipeline/README.md:23]: Providing param values in this format is quite difficult. Is it possible for the task to generate this json in the task body based on an array of key-values?

```
params:
  - name: ARGS
    value:
        - id=123
        - appName=foo
```

File: trigger-jenkins-pipeline/trigger-jenkins-pipeline.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: trigger-jenkins-pipeline
```
Comment (siamaksade) [inline trigger-jenkins-pipeline/trigger-jenkins-pipeline.yaml:4]: `trigger-jenkins-job` might be a more suitable name as not all Jenkins jobs are pipelines.

**Comment by bobcatfish** [comment]
neat!!

**Comment by vinamra28** [comment]
@ImJasonH Thanks for the review. Sure, will love to do that and take this task to the higher level. :smile:

**Comment by vinamra28** [comment]
hi @ImJasonH, what can I do to get this PR merged and start with the Custom Task proposal or are we going to include that(Custom Task Proposal) in the current PR. Also please guide me as how to get started with that :smile:

**Comment by ImJasonH** [comment]
> hi @ImJasonH, what can I do to get this PR merged and start with the Custom Task proposal or are we going to include that(Custom Task Proposal) in the current PR. Also please guide me as how to get started with that 😄

Hey, sorry for the delay. I don't think we should drop this PR in favor of a Custom Tasks approach, so submitting it as a regular Task like you're doing here should be fine. The only problem is I actually have no familiarity with Jenkins and so probably can't really review it very well. Is there someone else you think could review it that might know more about Jenkins?

File: task/trigger-jenkins-job/0.1/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
+# Trigger Jenkins Job
+
+The following task can be used to trigger a Jenkins pipeline using CURL request from a Tekton Task.
+
+More details on Remote Access API can be found [here](https://www.jenkins.io/doc/book/using/remote-access-api/)
+
+## Install the Task
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/trigger-jenkins-job/trigger-jenkins-job.yaml
```
Comment (PuneetPunamiya) [inline task/trigger-jenkins-job/0.1/README.md:10]: I think the path needs to be corrected, version directory is missing.

It should be :

`kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml`

**Comment by waveywaves** [comment]
@ImJasonH I just went through the manisfests and tried it out on Jenkins with @vinamra28. It looks good to me. It is definitely useful to have this to allow tekton users to trigger their jenkins jobs.

**Pull request merged**